### PR TITLE
Configure basepath through default configuration

### PR DESCRIPTION
**Pull request by soc-martin**
Related to issue #883

### CODE REVIEW

File: src/NLog.Web.AspNetCore/Config/SetupBuilderExtensions.cs [hunk 1]
```diff
@@ -25,7 +25,12 @@ public static ISetupBuilder LoadConfigurationFromAppSettings(this ISetupBuilder
             var currentBasePath = basePath;
             if (currentBasePath is null)
             {
-                currentBasePath = AppContext.BaseDirectory;
+                var webHostDefaultConfig = new ConfigurationBuilder()
+                    .AddEnvironmentVariables(prefix: "ASPNETCORE_")
+                    .Build();
+
+                var contentRootPath = webHostDefaultConfig.GetValue<string>(Microsoft.AspNetCore.Hosting.WebHostDefaults.ContentRootKey);
+                currentBasePath = ResolveContentRootPath(contentRootPath, AppContext.BaseDirectory);
```
Comment (snakefoot) [inline src/NLog.Web.AspNetCore/Config/SetupBuilderExtensions.cs:33]: What happens if you use this instead:
```c#
if (currentBasePath is null)
{
    currentBasePath = currentBasePath ?? GetAspNetCoreEnvironment("ASPNETCORE_CONTENTROOT") ?? GetAspNetCoreEnvironment("DOTNET_CONTENTROOT") ?? AppContext.BaseDirectory;
    if (string.IsNullOrEmpty(currentBasePath))
    {
       currentBasePath = Directory.GetCurrentDirectory();
    }
}
```

Comment (soc-martin) [reply src/NLog.Web.AspNetCore/Config/SetupBuilderExtensions.cs:33]: The environment variables for `ASPNETCORE_CONTENTROOT` and `DOTNET_CONTENTROOT` are not always set. I tested locally with an ASP.Net Core project of mine.

Comment (soc-martin) [reply src/NLog.Web.AspNetCore/Config/SetupBuilderExtensions.cs:33]: It's true that the environment variables should override the configuration though. I can update the PR with that in mind.

Comment (snakefoot) [reply src/NLog.Web.AspNetCore/Config/SetupBuilderExtensions.cs:33]: Tried calling this method:
```
currentBasePath = new ConfigurationBuilder().AddEnvironmentVariables(prefix: "ASPNETCORE_").Build().GetValue<string>(Microsoft.AspNetCore.Hosting.WebHostDefaults.ContentRootKey);
``
And can confirm that your method is also not working

Comment (soc-martin) [reply src/NLog.Web.AspNetCore/Config/SetupBuilderExtensions.cs:33]: Sorry for this inconvenience. I'll try and find an alternative solution in another PR.

Comment (snakefoot) [reply src/NLog.Web.AspNetCore/Config/SetupBuilderExtensions.cs:33]: But I do like including environment-variable `ASPNETCORE_CONTENTROOT`, so will include that.

**Comment by snakefoot** [comment]
@soc-martin Created #885 that prioritize current-directory, when it makes sense

**Comment by snakefoot** [comment]
Closing as superseeded by #885

**Pull request closed without merging**